src/pages/Settings.jsx

```diff
@@ -1,128 +1,256 @@
-  const [settingsFeedback, setSettingsFeedback] = useState(null);
-
   const containerStyle = {
     marginLeft: theme.margins.sideBarMargin,
     marginTop: theme.margins.headerMargin,
   };
 
+  const [password, setPassword] = useState('');
+  // Initialize alert preferences to false for the form
+  const [alertPreferences, setAlertPreferences] = useState({
+    Email: false,
+    Slack: false,
+    InApp: false,
+  });
+  // Provide feedback to the user once response is received from database
+  const [settingsFeedback, setSettingsFeedback] = useState(null);
+  // Slack URL and Email are required if the user has selected the checkboxes for them
+  const isSlackURLRequired = alertPreferences['Slack'];
+  const isEmailRequired = alertPreferences['Email'];
+  // Store previously saved preferences, updated when userInfo is retrieved from global state or settingsFeedback is updated after form submission
+  const [savedPreferences, setSavedPreferences] = useState({});
+
   const handlePasswordChange = (event) => setPassword(event.target.value);
 
-  const handleAlertChange = (type) => setAlertPreferences((prev) => ({ ...prev, [type]: !prev[type] }));
+  const handleAlertChange = (type) => {
+    setAlertPreferences((prev) => ({ ...prev, [type]: !prev[type] }));
+  };
+
+  // Use global context to access user info
+  const { userInfo, updateUserInfo } = useAppContext();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch(`${apiUrl}/alert/get-preferences/${userInfo.userID}`, {
+          method: 'GET',
+          headers: { 'Content-Type': 'application/json' },
+        });
+  
+        if (response.ok) {
+          const preferencesData = await response.json();
+          setSavedPreferences(preferencesData.data.preferences);
+          // console.log('Alert preferences received.');
+        }
+      } catch (error) {
+        console.error('Error while getting saved alert preferences:', error);
+      }
+    };
+  
+    // Fetch data when user is on Settings page. Optional chaining to allow reading the value of userInfo.userID even if userInfo is null or undefined without causing an error.
+    if (userInfo?.userID) {
+      fetchData();
+    }
+  }, [userInfo?.userID, settingsFeedback]);
+  
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
 
-  const saveAlertPreferences = async () => {
+    // Check to see if any changes were made before API call
+    const changesMade =
+      JSON.stringify(alertPreferences) !== JSON.stringify({
+        Email: false,
+        Slack: false,
+        InApp: false,
+      });
+
+    if (!changesMade) {
+      setSettingsFeedback('No changes were made.');
+      return;
+    }
+
+    if (
+      !alertPreferences.Email &&
+      !alertPreferences.Slack &&
+      !alertPreferences.InApp
+    ) {
+      setSettingsFeedback('Select at least one alert preference.');
+      return;
+    }
+
+    if (!password.trim()) {
```

Comment:
Doesn't check entered password with hashed password in DB

Reply:
updated